### DISCUSSION TITLE
Skip Teardown of Tests(Including fixtures it uses)

### DISCUSSION THREAD

**Discussion by taylornielson22** [pytest-dev/pytest#10964] [Q&A]
How do I skip teardown phase of test(including fixtures it uses) in pytest? I have seen people suggest adding finalizers to fixtures, but **we want to automatically skip teardown of fixtures created outside of our plugin.**

**Comment by RonnyPfannschmidt** [comment]
As no surrounding context Was provided, all I can say that by design this currently is not possible in any sensible way and I'm not comfortable devising a hack to do it

**Reply by taylornielson22** [reply]
in our testing, we want to maintain failure state when test fail to debug the state of devices under test. to do that, we have to skip the teardown of the test which includes teardown in all fixture to achieve this. our fixture uses yield to perform setup teardown. we also need skip teardown feature, when we want to reproduce the issue for debugging. so that devices can stay in failure state for debugging. @RonnyPfannschmidt

**Reply by RonnyPfannschmidt** [reply]
My recommendation for that is to employ a wrapper object that disengages from the device and triggers pytest.xfail exceptions in the teardown functions so that the fixtures will tear down with a expected failure

A alternative could be a sys._exit, which is sudden process death without any cleanup (deadpool solution)

**Reply by taylornielson22** [reply]
@RonnyPfannschmidt We use the` pytest_runtest_teardown(item, nextitem)`  hook. In this hook, is there a way to skip run of other  `pytest_runtest_teardown(item, nextitem)`  hook function if some CLI argument is set?

Example workflow

Our plugin.py
```python
def pytest_addoption(parser):
  parser.addoption(
        "--skip-test-teardown",
        action="store_true",
        dest="skipteardown",
        default=False,
    )

def pytest_runtest_teardown(item, nextitem):
  if item.config.skipteardown:
      # Skip test teardown somehow

@pytest.fixture()
def logger():
  logger = logging.getLogger("logger")
  yield logger
  logger.info("In Teardown")
```

Other teams that use our plugin test file
```python
@pytest.fixture()
def other_team_fixture(logger)
  yield 1
  logger.info("In Teardown")

def test_example(other_team_fixture):
  assert other_team_fixture
```

If `test_example` was ran with `--skip-test-teardown`, then the plugin's `logger` and the `other_team_fixture` fixture should not execute `logger.info("In Teardown")`

**Reply by RonnyPfannschmidt** [reply]
That's not possible safely

**Reply by taylornielson22** [reply]
I think pytest should consider adding a safe way to do this(if possible). I would think this is not an uncommon request. But thanks for your help and insight!

**Reply by RonnyPfannschmidt** [reply]
The pattern I recommended is the way to go
